ugs-core/src/com/willwinder/universalgcodesender/types/GcodeCommand.java

```diff
@@ -27,28 +19,40 @@ This file is part of Universal Gcode Sender (UGS).
 
 package com.willwinder.universalgcodesender.types;
 
-import com.willwinder.universalgcodesender.GrblUtils;
 import com.willwinder.universalgcodesender.gcode.GcodePreprocessorUtils;
+import org.apache.commons.lang3.builder.EqualsBuilder;
+import org.apache.commons.lang3.builder.HashCodeBuilder;
+import org.apache.commons.lang3.builder.ToStringBuilder;
+import org.apache.commons.lang3.builder.ToStringStyle;
+
+import java.util.concurrent.atomic.AtomicInteger;
+
 
 /**
+ * An object representing a single GcodeCommand.
  *
  * @author wwinder
  */
 public class GcodeCommand {
+    private static final AtomicInteger ID_GENERATOR = new AtomicInteger(0);
     private String command;
     private String originalCommand;
     private String response;
-    private String responseType;
     private Boolean sent = false;
-    private Boolean done = false;
     private Boolean isOk = false;
     private Boolean isError = false;
+    // TODO: Get rid of the commandNum member which is used in the GUI for indexing a table of these objects.
```

Comment:
I think that the command num is also used by gcode processors. For example the "arc expander" will split an arc into potentially hundreds of small line segments, each of which will retain the same command number. This way if a program were to fail, we could theoretically trace the program back to a line number in the file.

The gcode editor also used this to let you click on the arc command and highlight all of the line segments in the visualizer.

Reply:
I think it's a good idea to still have the command number, I just moved the TODO-comment from the file header thinking that you might have a bigger plan. I'll remove the TODO.